realm/realm-library/src/main/cpp/io_realm_internal_Collection.cpp

```diff
@@ -414,8 +414,46 @@ JNIEXPORT jboolean JNICALL Java_io_realm_internal_Collection_nativeDeleteFirst(J
     return JNI_FALSE;
 }
 
-JNIEXPORT void JNICALL Java_io_realm_internal_Collection_nativeDelete(JNIEnv* env, jclass, jlong native_ptr,
-                                                                      jlong index)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+JNIEXPORT void JNICALL Java_io_realm_internal_Collection_nativeSetNull(JNIEnv* env, jclass, jlong native_ptr, jlong index)
```

Comment:
I think there is a pretty good argument for keeping any checks that are reasonable, on this side of the JNI boundary.